src/go/k8s/pkg/resources/statefulset_update.go

```diff
@@ -264,39 +352,78 @@ func (r *StatefulSetResource) podEviction(ctx context.Context, pod, artificialPo
 		return err
 	}
 
-	var ordinal int64
+	var ordinal int32
 	ordinal, err = utils.GetPodOrdinal(pod.Name, r.pandaCluster.Name)
 	if err != nil {
 		return fmt.Errorf("cluster %s: cannot convert pod name (%s) to ordinal: %w", r.pandaCluster.Name, pod.Name, err)
 	}
 
-	if patchResult.IsEmpty() {
-		if err = r.checkMaintenanceMode(ctx, int32(ordinal)); err != nil {
+	managedDecommission, err := r.IsManagedDecommission()
+	if err != nil {
+		log.Error(err, "not performing a managed decommission")
+	}
+
+	isRedpandaConfigUpdate := utils.IsStatusPodConditionTrue(pod.Status.Conditions, ClusterUpdatePodCondition)
+
+	// if the pod definition changes, patchResult is not empty
+	// if this isn't a managedDecommission
+	// if the pod isn't needing to be updated for redpanda
+	// take the pod out of maintenance mode
+	if patchResult.IsEmpty() && !managedDecommission && !isRedpandaConfigUpdate {
+		if err = r.checkMaintenanceMode(ctx, ordinal); err != nil {
 			return &RequeueAfterError{
 				RequeueAfter: RequeueDuration,
 				Msg:          fmt.Sprintf("checking maintenance node (%s): %v", pod.Name, err),
 			}
 		}
-		return nil
+		return &RequeueAfterError{RequeueAfter: RequeueDuration, Msg: "wait for pod restart"}
 	}
 
-	if *r.pandaCluster.Spec.Replicas > 1 {
-		r.logger.Info("Put broker into maintenance mode",
+	if *r.pandaCluster.Spec.Replicas == 1 {
+		log.Info("Changes in Pod definition other than activeDeadlineSeconds, configurator and Redpanda container name. Deleting pod",
 			"pod-name", pod.Name,
 			"patch", patchResult.Patch)
-		if err = r.putInMaintenanceMode(ctx, int32(ordinal)); err != nil {
-			// As maintenance mode can not be easily watched using controller runtime the requeue error
-			// is always returned. That way a rolling update will not finish when operator waits for
-			// maintenance mode finished.
-			return &RequeueAfterError{
-				RequeueAfter: RequeueDuration,
-				Msg:          fmt.Sprintf("putting node (%s) into maintenance mode: %v", pod.Name, err),
-			}
+
+		if err = r.Delete(ctx, pod); err != nil {
+			return fmt.Errorf("unable to remove Redpanda pod: %w", err)
+		}
+		return nil
+	}
+	if managedDecommission {
+		log.Info("managed decommission is set: decommission broker")
+		var id *int32
+		if id, err = r.getBrokerIDForPod(ctx, ordinal); err != nil {
+			return err
+		}
+		r.pandaCluster.SetDecommissionBrokerID(id)
+
+		if err = r.handleDecommission(ctx, log); err != nil {
+			return err
```

Comment:
NIT: Add context to the error?

Reply:
handleDecommission is already quite clear in its errors. I don't think I can add any context of value here.